### DISCUSSION TITLE
Need Help | Not found in Doc | Optional fields

### DISCUSSION THREAD

**Discussion by nmaqsood91** [colinhacks/zod#1940] [Q&A] [answered]
Hello 

I have a scenario, I have 2 fields and both of the fields are optional, the tricky thing user can leave both of the fields blank but if the user wants to input he can give value to one of the fields only

```ts
const paymentObject = z.object({
  creditorAccountIban: z.string().min(15).max(34),
  description: z.string().optional(),
  paymentReference: z.string().optional()
});
```

so description and paymentReference can be empty, but if the user wants to input only one can have a value either description or paymentReference.

Please give me some good examples

**Comment by JacobWeisenburger** [comment, marked as answer]
https://github.com/colinhacks/zod#optionals

Your schema seems to work as you have described. Is there something that isn't working as you expect?

Here's an example of how optional works:
```ts
const schema = z.object({
    foo: z.string().optional(),
    bar: z.string().optional(),
})

console.log(schema.safeParse({ foo: 'foo', bar: 'bar' }).success) // true
console.log(schema.safeParse({ foo: 'foo' }).success) // true
console.log(schema.safeParse({ bar: 'bar' }).success) // true
console.log(schema.safeParse({}).success) // true

console.log(schema.safeParse(undefined).success) // false
console.log(schema.safeParse('string').success) // false
console.log(schema.safeParse(42).success) // false
```
